src/test/java/com/arpnetworking/kairosdb/aggregators/HistogramFilterAggregatorTest.java

```diff
@@ -0,0 +1,871 @@
+/**
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.kairosdb.aggregators;
+
+import com.arpnetworking.kairosdb.HistogramDataPoint;
+import com.arpnetworking.kairosdb.HistogramDataPointImpl;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.kairosdb.core.DataPoint;
+import org.kairosdb.core.aggregator.FilterAggregator;
+import org.kairosdb.core.datapoints.DoubleDataPoint;
+import org.kairosdb.core.datastore.DataPointGroup;
+import org.kairosdb.testing.ListDataPointGroup;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Unit tests for the Histogram Filter Aggregator.
+ *
+ * @author Joey Jackson
+ */
+public class HistogramFilterAggregatorTest {
```

Comment:
Great coverage! It is good to have tests for all of those fiddly nested ifs in `HistogramFilterAggregator`.
Two things that I think would help:

- I notice the following pattern recurring many times:

    ```java
    ListDataPointGroup foo = new ListDataPointGroup(...);
    final TreeMap<Double, Integer> bar = new TreeMap<>();
    bar.put(HistogramFilterAggregator.truncate(Double.longBitsToDouble(0x...L)), ...);
    ... // repeat for some number of bins
    foo.addDataPoint(createHistogram(t, bar));
    ... // repeat for some number of datapoints
    ```

    Could we encapsulate this in a method? I think this could make the tests a lot more concise.

- I notice there are just a handful of hex constants, each occurring many times -- for readability/robustness, could we refactor them into constants? Maybe something like

    ```java
        private static final Double _100_0 = Double.longBitsToDouble(0x4059000000000000L);
        private static final Double _100_01 = Double.longBitsToDouble(0x40590000FFFFFFFFL);
        ...
    ```

In conjunction, I think these would make each unit test _way_ simpler, so each test would be as simple as

```java
@Test
public void testFilterLessThanDiscardThresholdAtBinBoundaryPositiveBins() {
     runTest(
         FilterAggregator.FilterOperation.LT, HistogramFilterAggregator.FilterIndeterminate.DISCARD,
            true, true,
        makeGroup(listOf(_100_0, _100_5), listOf(_1e79, _1e79))
    );
}
```
which would make it much more intuitively obvious what assertions each test was making.